java-frontend/src/main/java/org/sonar/java/model/ExpressionUtils.java

```diff
@@ -33,12 +36,50 @@ private ExpressionUtils() {
    * In case of simple assignments, only the expression is evaluated, as we only use the reference to the variable to store the result.
    * For SE-Based checks, only a single value should be unstacked if its the case. For other cases, two values should be unstacked.
    * See JLS8-15.26
-   * 
+   *
    * @param tree The assignment tree
-   * @return true if the tree is a simple assignment 
+   * @return true if the tree is a simple assignment
+   * @see #extractIdentifier(AssignmentExpressionTree)
    */
   public static boolean isSimpleAssignment(AssignmentExpressionTree tree) {
-    return tree.is(Tree.Kind.ASSIGNMENT) && ExpressionUtils.skipParentheses(tree.variable()).is(Tree.Kind.IDENTIFIER);
+    if (!tree.is(Tree.Kind.ASSIGNMENT)) {
+      // This can't possibly be a simple assignment.
+      return false;
+    }
+
+    ExpressionTree variable = ExpressionUtils.skipParentheses(tree.variable());
+    return variable.is(Tree.Kind.IDENTIFIER) || (variable.is(Tree.Kind.MEMBER_SELECT) && isThisAssignment((MemberSelectExpressionTree) variable));
+  }
+
+  public static boolean isSimpleAssignment(MemberSelectExpressionTree memberSelectExpressionTree) {
+      return isThisAssignment(memberSelectExpressionTree);
+  }
+
+  public static IdentifierTree extractIdentifier(AssignmentExpressionTree tree) {
+    ExpressionTree variable = skipParentheses(tree.variable());
+    if (variable.is(Tree.Kind.IDENTIFIER)) {
+      return (IdentifierTree) variable;
+    }
+
+    if (variable.is(Tree.Kind.MEMBER_SELECT)) {
+      MemberSelectExpressionTree selectTree = (MemberSelectExpressionTree) variable;
+      if (isThisAssignment(selectTree)) { // FIXME(Johan): Is this check needed?
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)